handlers/graph_unused.go

```diff
@@ -0,0 +1,145 @@
+package handlers
+
+import (
+	"fmt"
+
+	"k8s.io/api/apps/v1beta1"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/graph/tree"
+	"github.com/kiali/kiali/log"
+)
+
+func addUnusedNodes(trees *[]tree.ServiceNode, namespaceName string, deployments *v1beta1.DeploymentList) {
+	staticNodeList := buildStaticNodeList(namespaceName, deployments)
+	currentNodeSet := make(map[string]struct{})
+	for _, tree := range *trees {
+		buildNodeSet(&currentNodeSet, &tree)
+	}
+
+	// Empty trees, no traffic in whole namespace, so we create a default tree with the static info
+	if len(currentNodeSet) == 0 {
+		buildDefaultTrees(trees, &staticNodeList)
+		return
+	}
+
+	// There is traffic in the namespace, so we need to check if we have nodes without traffic
+	for i := 0; i < len(staticNodeList); i++ {
+		// Node found in the static list but with no traffic, it should be added to the trees
+		if _, ok := currentNodeSet[staticNodeList[i].ID]; !ok {
+			addNodeToTrees(trees, &staticNodeList[i])
+		}
+	}
+}
+
+func buildStaticNodeList(namespaceName string, deployments *v1beta1.DeploymentList) []tree.ServiceNode {
+	nonTrafficList := make([]tree.ServiceNode, 0)
+	appLabel := config.Get().ServiceFilterLabelName
+	versionLabel := config.Get().VersionFilterLabelName
+	identityDomain := config.Get().IstioIdentityDomain
+	for _, deployment := range deployments.Items {
+		app, ok := deployment.GetObjectMeta().GetLabels()[appLabel]
+		if !ok {
+			log.Warningf("Deployment %s has not a proper app label [%s]", deployment.Name, appLabel)
+			continue
+		}
+		version, ok := deployment.GetObjectMeta().GetLabels()[versionLabel]
+		if !ok {
+			log.Warningf("Deployment %s has not a proper version label [%s]", deployment.Name, versionLabel)
+			continue
+		}
+		staticNode := tree.NewServiceNode(fmt.Sprintf("%s.%s.%s", app, namespaceName, identityDomain), version)
```

Comment:
The first "%s" is the istio service name, is it assumed to be the same as the app label value? I wonder if we could end with the same problem @xeviknal is hitting for matching deployments and services, which is that we can't rely on such labels.

Reply:
As these ones are Istio services, I think we are safe.